### DISCUSSION TITLE
Where to put test files

### DISCUSSION THREAD

**Discussion by Davy238** [community/community#102622] [New to GitHub]
I am new to Git and finding my way.
After several false starts, I have now put all 20 of my scripts for all programming languages and also modules in the main folder and made it Git controlled. I regard this as my folder of operational tested scripts. I dont want to work on these because they may need to be used at any time - dont want work in progress. So I never edit the main branch.
I have created another branch which I call 'Dev' and Git automatically populated it with all the 20 script files from Main. It is here that I do all my code development, and of course, when finished merge the whole of Dev back into the main (operational) branch. Whoops, writing that I have just reliased that some part completed scripts might over write the operational equivalent!
My question is two-fold;

- Is the above a reasonable thing to do?

- Where do I put my (small) test files?

If I put them in the repository folder then they show up in the main/operational branch and it gets messy.
I could add the test files to the repository, make them controlled in the 'Dev' branch then ReMove them from the Main branch.  But then in the Dev branch I am likely to have a mix of test files for the different scripts in Dev.
Or should all my 30 scripts each have their own dev branch; containing their test files?

Usually when I have these problems it is because I have misunderstood how Git works.

Any thoughts?

**Comment by DJLink** [comment]
1 - yes

2 -this is more or an organizational thing and how it works best for you, there is really no wrong way here if it works for you.
you can keep your tests under a /tests folder for example.
If you are talking about tests as in changes for the files on Main branch then that separate "dev" branch is the way to go, if you ever feel like you wanna merge you can, otherwise just leave it separate.

Another thing you can consider is adding the test files as submodule https://github.blog/2016-02-01-working-with-submodules/ on your repo, that way it's on the folder but really on a separate repo and you can update it freely as well.
